auction-app-backend/src/main/java/com/example/auctionapp/service/implementation/ProductServiceImpl.java

```diff
@@ -278,6 +278,19 @@ public List<Product> getRandomProductsByCategoryId(final UUID categoryId, final
                 .toList();
     }
 
+    @Override
+    public List<Product> uploadProducts(final MultipartFile file, final UUID userId) throws IOException {
+        final CsvUtil csvUtil = new CsvUtil();
+
+        final UserEntity user = this.userRepository.findById(userId)
+                .orElseThrow(() -> new ResourceNotFoundException("User with the given ID does not exist"));
+
+        return this.productRepository.saveAll(csvUtil.uploadProduct(file, user, categoryRepository, productImageRepository, productRepository))
```

Comment:
One thing I don't like here is that we directly pass csvUtil method to the saveAll
It would be much cleaner to store the result of the csvUtil method in a variable(list) and that pass to the method